backend/src/DataSource/MqttSource.ts

```diff
@@ -98,7 +98,7 @@ export class MqttSource implements DataSource<MqttOptions> {
 
   public publish(msg: MqttMessage) {
     if (this.client) {
-      this.client.publish(msg.topic, msg.payload ? Base64Message.toUnicodeString(msg.payload) : '', {
+      this.client.publish(msg.topic, msg.payload?.toBuffer() ?? '', {
```

Comment:
You are right, this is a bit messy.
I'll add a DTO (Data Transfer Object) type to differentiate between these two types.

Reply:
> You are right, this is a bit messy. I'll add a DTO (Data Transfer Object) type to differentiate between these two types.

Good, publish is broken at the moment

Reply:
Nice catch btw., Maybe I'll add a few automated test scenarios later on.

This should work now.

Reply:
> Nice catch btw., Maybe I'll add a few automated test scenarios later on.
> 
> This should work now.

Speaking of automated tests, I did a dirty hack when moving to Playwright, the expandTopic logic is broken. It does not work when topics have the same name in path since multiple nodes will be matched.

i.e

`/livingtroom/lamp`
`/bedroom/upstairs/lamp`
`/lamp/device1`

I opened an issue on this https://github.com/thomasnordquist/MQTT-Explorer/issues/796